cpp/src/arrow/type_traits.h

```diff
@@ -1624,6 +1639,16 @@ static inline bool is_binary(const DataType& type) { return is_binary(type.id())
 /// Convenience for checking using the type's id
 static inline bool is_string(const DataType& type) { return is_string(type.id()); }
 
+/// \brief Check for a binary-view-like type
+///
+/// \param[in] type the type to check
+/// \return whether type is a binary-view-like type
+///
+/// Convenience for checking using the type's id
+static inline bool is_binary_view_like(const DataType& type) {
```

Comment:
By the way, should all the `static inline bool is_XXX(const DataType& type)` predicates be turned into `constexpr`? This probably needs `DataType::id()` to be made `constexpr` as well.

Reply:
I think there's no utility in constexpr type predicates for now. I have a proposal somewhere for an improved C++17 `visit` function where we might use constexpr type predicates:

```c++
const DataType& type = ...;
VisitType(type, [&](auto type) {
  if constexpr (is_binary_view_like(type)) {
    // access to type specific properties can be done here
    decltype(type)::TypeClass::is_utf8;
  }
});
```

Reply:
What do you mean "no utility"? It is stricly better to make them `constexpr`, even if not needed in our own code, AFAICT.

Reply:
Important to notice that all the functions taking `DataType&` are not `constexpr`, so this is out-of-scope for this PR.

Reply:
Right, though it's probably an easy change that we can do at some point.